src/main/java/pl/psnc/dei/service/DDBRestService.java

```diff
@@ -0,0 +1,103 @@
+package pl.psnc.dei.service;
+
+import org.apache.jena.atlas.json.JSON;
+import org.apache.jena.atlas.json.JsonObject;
+import org.apache.jena.rdf.model.Model;
+import org.apache.jena.rdf.model.ModelFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.stereotype.Service;
+import org.springframework.web.reactive.function.client.WebClient;
+import org.w3c.dom.Document;
+import org.w3c.dom.Node;
+import org.xml.sax.InputSource;
+import org.xml.sax.SAXException;
+import pl.psnc.dei.exception.DEIHttpException;
+import pl.psnc.dei.request.RestRequestExecutor;
+import reactor.core.publisher.Mono;
+
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.parsers.ParserConfigurationException;
+import javax.xml.transform.OutputKeys;
+import javax.xml.transform.Transformer;
+import javax.xml.transform.TransformerException;
+import javax.xml.transform.TransformerFactory;
+import javax.xml.transform.dom.DOMSource;
+import javax.xml.transform.stream.StreamResult;
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.StringReader;
+import java.io.StringWriter;
+import java.nio.charset.StandardCharsets;
+
+@Service
+public class DDBRestService extends RestRequestExecutor {
+
+	private final Logger logger = LoggerFactory.getLogger(DDBRestService.class);
+
+	@Value("${ddb.api.items}")
+	private String ddbApiItems;
+
+	@Value("${ddb.api.key}")
+	private String oauth_key;
+
+	public DDBRestService(WebClient.Builder webClientBuilder) {
+		configure(webClientBuilder);
+	}
+
+	public JsonObject retrieveRecordFromEuropeanaAndConvertToJsonLd(String recordId) {
+		logger.info("Retrieving record from ddb {}", recordId);
+
+		String xmlRecord = webClient.get()
+				.uri(b -> b.path(ddbApiItems).queryParam("oauth_consumer_key", oauth_key).build(recordId))
+				.accept(MediaType.APPLICATION_XML)
+				.retrieve()
+				.onStatus(HttpStatus::is4xxClientError, clientResponse -> Mono.error(new DEIHttpException(clientResponse.rawStatusCode(), clientResponse.statusCode().getReasonPhrase())))
+				.onStatus(HttpStatus::is5xxServerError, clientResponse -> Mono.error(new DEIHttpException(clientResponse.rawStatusCode(), clientResponse.statusCode().getReasonPhrase())))
+				.bodyToMono(String.class)
+				.block();
+
+		return convertToJsonLd(xmlRecord);
+	}
+
+	private JsonObject convertToJsonLd(String record) {
```

Comment:
I tried this(after fixing request), and i got some exception, so try to test it.

Reply:
what error did you get? There could be an error when you try to return JsonObject because it has complicated structure and can't be serialized. When you would get response directly from this method to test it you should add return ...toString() and change returning value for String

Reply:
Ahh, i should copy that stack... It could be the thing u mentioned, but i`m not sure.
You just need to call that method to see this stack.